From d5b43904699f6f27c4e5026828fb5e11f1bb74d0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 27 Feb 2024 14:24:27 +0100
Subject: [PATCH 632/841] feat(stm32mp2-fdts): add CRYP1 and CRYP2 to mp2
 device tree
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The stm32mp25xc or stm32mp25xf SoC have CRYP1 and CRYP2 IP available,
so add it in the corresponding device tree.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 494796ddf849cb575f7c6f3a4120dc71e0feb742 feat(stm32mp2-fdts): add CRYP1 and CRYP2 to mp2 device tree]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: I1bb027c10f16e5b2df1b193b3e8cea479f9ad96b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/363238
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/386065
---
 fdts/stm32mp25xc.dtsi | 16 ++++++++++++++++
 fdts/stm32mp25xf.dtsi | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/fdts/stm32mp25xc.dtsi b/fdts/stm32mp25xc.dtsi
index de85a9498..dd0036ca3 100644
--- a/fdts/stm32mp25xc.dtsi
+++ b/fdts/stm32mp25xc.dtsi
@@ -6,6 +6,22 @@
 
 / {
 	soc@0 {
+		cryp1: cryp@42030000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42030000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP1>;
+			resets = <&rcc CRYP1_R>;
+			status = "disabled";
+		};
+
+		cryp2: cryp@42040000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42040000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP2>;
+			resets = <&rcc CRYP2_R>;
+			status = "disabled";
+		};
+
 		saes: saes@42050000 {
 			compatible = "st,stm32-saes";
 			reg = <0x42050000 0x400>;
diff --git a/fdts/stm32mp25xf.dtsi b/fdts/stm32mp25xf.dtsi
index de85a9498..dd0036ca3 100644
--- a/fdts/stm32mp25xf.dtsi
+++ b/fdts/stm32mp25xf.dtsi
@@ -6,6 +6,22 @@
 
 / {
 	soc@0 {
+		cryp1: cryp@42030000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42030000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP1>;
+			resets = <&rcc CRYP1_R>;
+			status = "disabled";
+		};
+
+		cryp2: cryp@42040000 {
+			compatible = "st,stm32mp1-cryp";
+			reg = <0x42040000 0x400>;
+			clocks = <&rcc CK_BUS_CRYP2>;
+			resets = <&rcc CRYP2_R>;
+			status = "disabled";
+		};
+
 		saes: saes@42050000 {
 			compatible = "st,stm32-saes";
 			reg = <0x42050000 0x400>;
-- 
2.47.2

